generators/app/templates/gulp/tasks/templates.js

```diff
@@ -1,12 +1,31 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var templatesTask = function (gulp, plugins, config, helpers) {
 
+  function getTemplateInputData(file) {
+    try {
+      var data = JSON.parse(
+        fs.readFileSync(config.src.dataPath + path.basename(file.path) + '.json')
+      );
+    } catch(e) {
```

Comment:
Thought we're having JSON files here too. But as we discussed with @jakub300 there would be a problem in such case, because we would not be sure which one (.json or .twig) comes first. 
One possible optimisation here would be to check if `src/data` exists and then perform the file read.